nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/util/TlsHelper.java

```diff
@@ -0,0 +1,177 @@
+        this(createKeyPairGenerator(keyPairAlgorithm, keySize), days, signingAlgorithm);
+    }
+
+    protected TlsHelper(KeyPairGenerator keyPairGenerator, int days, String signingAlgorithm) {
+        this.keyPairGenerator = keyPairGenerator;
+        this.days = days;
+        this.signingAlgorithm = signingAlgorithm;
+    }
+
+    public static void addBouncyCastleProvider() {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    private static KeyPairGenerator createKeyPairGenerator(String algorithm, int keySize) throws NoSuchAlgorithmException {
+        KeyPairGenerator instance = KeyPairGenerator.getInstance(algorithm);
+        instance.initialize(keySize);
+        return instance;
+    }
+
+    public KeyPair generateKeyPair() throws NoSuchAlgorithmException {
+        return keyPairGenerator.generateKeyPair();
+    }
+
+    public void addToKeyStore(KeyStore keyStore, KeyPair keyPair, String alias, char[] passphrase, Certificate... certificates) throws GeneralSecurityException, IOException {
+        keyStore.setKeyEntry(alias, keyPair.getPrivate(), passphrase, certificates);
+    }
+
+    public KeyStore createKeyStore(String keyStoreType) throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException, NoSuchProviderException {
+        KeyStore keyStore;
+        if (PKCS12.equals(keyStoreType)) {
+            keyStore = KeyStore.getInstance(keyStoreType, BouncyCastleProvider.PROVIDER_NAME);
+        } else {
+            keyStore = KeyStore.getInstance(keyStoreType);
+        }
+        keyStore.load(null, null);
+        return keyStore;
+    }
+
+    public X509Certificate generateSelfSignedX509Certificate(KeyPair keyPair, String dn) throws CertificateException {
+        return CertificateUtils.generateSelfSignedX509Certificate(keyPair, dn, signingAlgorithm, days);
+    }
+
+    public X509Certificate generateIssuedCertificate(String dn, PublicKey publicKey, X509Certificate issuer, KeyPair issuerKeyPair) throws CertificateException {
+        return CertificateUtils.generateIssuedCertificate(dn, publicKey, issuer, issuerKeyPair, signingAlgorithm, days);
+    }
+
+    public JcaPKCS10CertificationRequest generateCertificationRequest(String requestedDn, KeyPair keyPair) throws OperatorCreationException {
+        JcaPKCS10CertificationRequestBuilder jcaPKCS10CertificationRequestBuilder = new JcaPKCS10CertificationRequestBuilder(new X500Principal(requestedDn), keyPair.getPublic());
+        JcaContentSignerBuilder jcaContentSignerBuilder = new JcaContentSignerBuilder(signingAlgorithm);
+        return new JcaPKCS10CertificationRequest(jcaPKCS10CertificationRequestBuilder.build(jcaContentSignerBuilder.build(keyPair.getPrivate())));
+    }
+
+    public X509Certificate signCsr(JcaPKCS10CertificationRequest certificationRequest, X509Certificate issuer, KeyPair issuerKeyPair) throws InvalidKeySpecException, EACException,
```

Comment:
Collapsed everything in the helper that throws more than one type of security exception into GeneralSecurityException